src/base/datetime/calendar/header/dropdown/year/index.ts

```diff
@@ -0,0 +1,227 @@
+        @kuc:calendar-listbox-click="${this._handleChangeMenu}"
+        aria-hidden="${!this._menuVisible}"
+        ?hidden="${!this._menuVisible}"
+      >
+      </kuc-base-datetime-listbox>
+    `;
+  }
+
+  private _getStyleTagTemplate() {
+    return html`
+      <style>
+        .kuc-base-datetime-year-dropdown__toggle {
+          position: relative;
+          box-sizing: border-box;
+          height: 32px;
+          padding: 0 14px 0 8px;
+          line-height: 30px;
+          overflow: hidden;
+          background-color: white;
+          border: 1px solid transparent;
+          cursor: pointer;
+        }
+        .kuc-base-datetime-year-dropdown__toggle__icon {
+          flex: none;
+          width: 38px;
+          height: 38px;
+        }
+      </style>
+    `;
+  }
+
+  private _handleMouseUpDropdownToggle(event: MouseEvent) {
+    event.preventDefault();
+  }
+
+  private _handleMouseDownDropdownToggle(event: MouseEvent) {
+    event.preventDefault();
+  }
+
+  private _handleClickDropdownYearToggle(event: MouseEvent) {
+    if (!this._menuVisible) {
+      this._openMenu();
+    } else {
+      this._closeMenu();
+    }
+  }
+
+  private _handleKeyDownYearToggle(event: KeyboardEvent) {
+    if (!this._menuVisible) {
+      this._menuEl.highlightFirstItem();
+      return;
+    }
+    switch (event.key) {
+      case "Up":
+      case "ArrowUp": {
+        event.preventDefault();
+        this._menuEl.highlightPrevItem();
+        this._menuEl.scrollToView();
+        this._setActiveDescendant(
+          this._toggleEl,
+          this._menuEl.getHighlightItemId()
+        );
+        break;
+      }
+      case "Down":
+      case "ArrowDown": {
+        event.preventDefault();
+        this._menuEl.highlightNextItem();
+        this._menuEl.scrollToView();
+        this._setActiveDescendant(
+          this._toggleEl,
+          this._menuEl.getHighlightItemId()
+        );
+        break;
+      }
+      case "Home":
+        event.preventDefault();
+        this._menuEl.highlightFirstItem();
+        this._menuEl.scrollToTop();
+        this._setActiveDescendant(
+          this._toggleEl,
+          this._menuEl.getHighlightItemId()
+        );
+        break;
+      case "End":
+        event.preventDefault();
+        this._menuEl.highlightLastItem();
+        this._menuEl.scrollToBottom();
+        this._setActiveDescendant(
+          this._toggleEl,
+          this._menuEl.getHighlightItemId()
+        );
+        break;
+      case "Enter": {
+        event.preventDefault();
+        const highlightValue = this._menuEl.getHighlightValue();
+        if (highlightValue) {
+          this.year = Number(highlightValue);
+          const detail: CustomEventDetail = { value: `${this.year}` };
+          dispatchCustomEvent(this, "kuc:year-dropdown-change", detail);
+        }
+        this._menuVisible = false;
+        break;
+      }
+    }
+  }
+
+  private _handleBlurDropdownYearToggle(event: Event) {
```

Comment:
Delete the unnecessary Event.
```suggestion
  private _handleBlurDropdownYearToggle() {
```